velox/exec/HashBuild.cpp

```diff
@@ -1240,4 +1277,10 @@ void HashBuildSpiller::extractSpill(
         rows.data(), rows.size(), false, false, result->childAt(types.size()));
   }
 }
+
+bool HashBuild::abandonBuildNoDupHashEarly(int64_t numDistinct) const {
```

Comment:
Curious why do we abandon? Why not we always do the deduped way?

Reply:
@tanjialiang You can check the benchmark tests in #11212. When the duplicate rate of join keys in a left semi/anti join is not high, deduplication incurs additional overhead. Additionally, perhaps making this optimization disabled by default would make it more acceptable for the community to merge?